account.rs/src/wasm/src/account.rs

```diff
@@ -0,0 +1,345 @@
+        let secret_key = SecretKey::from(secret_key_bytes);
+        secret_key.public_key().as_bytes().to_vec()
+    }
+
+    #[wasm_bindgen(js_name = generateSecret)]
+    pub fn generate_secret(seed: &str, domain: &str) -> Vec<u8> {
+        let input = format!("{}{}", seed, domain);
+        blake2_string(input, ACCOUNT_HASH_LENGTH)
+    }
+
+    pub fn sign(&self, commitment: String, merkle_path: u32) -> Vec<u8> {
+        let inputs: Vec<Vec<u8>> = vec![
+            self.private_key.to_vec(),
+            commitment.into_bytes(),
+            merkle_path.to_be_bytes().to_vec(),
+        ];
+        poseidon_hash(inputs).unwrap()
+    }
+
+    #[wasm_bindgen(js_name = encryptAes)]
+    pub fn encrypt_aes(&self, message: Vec<u8>, iv16: Vec<u8>) -> Vec<u8> {
+        let key = Key::from_slice(&self.aes_secret);
+        let cipher = Aes256GcmSiv::new(key);
+        let nonce = Nonce::from_slice(&iv16);
+        let ciphertext = cipher
+            .encrypt(nonce, message.as_ref())
+            .expect("encryption failure!");
+
+        let mut result = iv16;
+        result.extend(ciphertext);
+        result
+    }
+
+    #[wasm_bindgen(js_name = decryptAes)]
+    pub fn decrypt_aes(&self, encrypted: Vec<u8>) -> Vec<u8> {
+        let iv16 = encrypted[0..16].to_vec();
+        let cipher_text = encrypted[16..].to_vec();
+        self.decrypt_aes_internal(cipher_text, iv16)
+    }
+
+    fn decrypt_aes_internal(&self, cipher_text: Vec<u8>, iv16: Vec<u8>) -> Vec<u8> {
+        let key = Key::from_slice(&self.aes_secret);
+        let cipher = Aes256GcmSiv::new(key);
+        let nonce = Nonce::from_slice(&iv16);
+        let decrypted_message = cipher
+            .decrypt(nonce, cipher_text.as_ref())
+            .expect("decryption failure!");
+        decrypted_message
+    }
+
+    #[wasm_bindgen(js_name = getUtxoPrefixViewingKey)]
+    pub fn get_utxo_prefix_viewing_key(&self, salt: &str) -> Vec<u8> {
+        if let Some(hashing_secret) = &self.hashing_secret {
+            let input = hashing_secret.to_vec();
+            let hashing_secret_string = String::from_utf8(input).unwrap();
+            Account::generate_secret(&hashing_secret_string, salt)
+        } else {
+            panic!("hashing_secret is None")
+        }
+    }
+
+    fn generate_utxo_prefix_hash(
+        &self,
+        merkle_tree_public_key: Vec<u8>,
+        prefix_counter: BigUint,
+        prefix_length: usize,
+    ) -> Vec<u8> {
+        let mut input = self.get_utxo_prefix_viewing_key("hashing");
+        input.extend(merkle_tree_public_key);
+        input.extend(prefix_counter.to_bytes_be());
+        blake2(&input, prefix_length)
+    }
+
+    #[wasm_bindgen(js_name = generateLatestUtxoPrefixHash)]
+    pub fn generate_latest_utxo_prefix_hash(
+        &mut self,
+        merkle_tree_public_key: Vec<u8>,
+        prefix_length: usize,
+    ) -> Vec<u8> {
+        let hash = self.generate_utxo_prefix_hash(
+            merkle_tree_public_key,
```

Comment:
And I would do conversion from vec to slice here. This way, we are also going to validate that the length of provided pubkey is as expected.